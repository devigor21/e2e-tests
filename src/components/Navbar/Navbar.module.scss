$grey: #dddddd;
$dark-grey:rgb(70, 70, 70);
$pink: #be81e7;
$blue: #236fc5;
$whity: #F9F3F3;

@mixin hoverColor() {
  background-color: $whity;
  color: $blue;
}

.nav {
  display: flex;
  justify-content: center;
  align-items: center;
  background: $blue;
  font-weight: 700;
  height: 3.5rem;
  width: 100vw;
  position: fixed;
  z-index: 101;
  top: 0;
  left: 0;
  font-size: 1rem;
  color: $whity;

  &>*:hover {
    cursor: pointer;
  }

  &__wrapper {
    display: flex;
    align-items: center;
    justify-content: right;
    color: inherit;
    text-decoration: none;
  }

  //logo
  &__logo {
    position: absolute;
    display: flex;
    align-items: center;
    font-size: 2rem;
    padding-left: 1.2rem;
    left: 0;
    z-index: 150;

    @media (max-width:650px) {
      display: none;
    }
  }

  &__logoTitle {
    font-size: 1.5rem;
    font-weight: 900;
    padding-left: 10px;

    &:hover {
      text-decoration: underline;
    }
  }

  //list
  &__list {
    display: flex;
    justify-content: center;
    flex-direction: row;
    z-index: 200;
    font-size: 1.2rem;
    padding: 0;
  }

  &__listItem {
    padding: 0 1.5rem;
    list-style-type: none;

    & a {
      text-decoration: none;
      color: white;

      &:hover {
        @include hoverColor();
      }
    }

    &:hover {
      @include hoverColor();
      cursor: initial;
    }
  }
}
