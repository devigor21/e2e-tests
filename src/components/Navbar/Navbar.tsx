import React from 'react';
import { NavLink } from 'react-router-dom';

import s from './Navbar.module.scss';

function Navbar() {
  return (
    <nav className={s.nav}>
      <NavLink to="/" className={s.nav__wrapper}>
        <div className={s.nav__logo}>
          <span className={s.nav__logoTitle}>TEST API</span>
        </div>
      </NavLink>

      <ul className={s.nav__list}>
        <li className={s.nav__listItem}>
          <NavLink to="/e2e-tests/about">About</NavLink>
        </li>
        <li className={s.nav__listItem}>
          <NavLink to="/e2e-tests">Home</NavLink>
        </li>
      </ul>
    </nav>
  );
}
export default Navbar;
