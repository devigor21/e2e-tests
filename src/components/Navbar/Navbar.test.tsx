import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import Navbar from '.';

describe('Navbar component', () => {
  it('should not change', () => {
    const navbar = render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(navbar).toMatchSnapshot();
  });
  it('should not set error', () => {
    const post = () => render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(post).not.toThrow();
  });
  it('should have About', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(screen.getByText('About')).toBeInTheDocument();
  });
});
