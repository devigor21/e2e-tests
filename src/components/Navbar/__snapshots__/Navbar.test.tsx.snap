// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should not change 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav"
      >
        <a
          aria-current="page"
          class="nav__wrapper active"
          href="/e2e-tests/"
        >
          <div
            class="nav__logo"
          >
            <span
              class="nav__logoTitle"
            >
              TEST API
            </span>
          </div>
        </a>
        <ul
          class="nav__list"
        >
          <li
            class="nav__listItem"
          >
            <a
              href="/e2e-tests/about"
            >
              About
            </a>
          </li>
          <li
            class="nav__listItem"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav"
    >
      <a
        aria-current="page"
        class="nav__wrapper active"
        href="/e2e-tests/"
      >
        <div
          class="nav__logo"
        >
          <span
            class="nav__logoTitle"
          >
            TEST API
          </span>
        </div>
      </a>
      <ul
        class="nav__list"
      >
        <li
          class="nav__listItem"
        >
          <a
            href="/e2e-tests/about"
          >
            About
          </a>
        </li>
        <li
          class="nav__listItem"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
